File: SingleNumber.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) 
    {
        int answer;
        
        int size = nums.size();
        
        if(size == 0)
        {
            return 0;
        }
        
        for(int i = 0; i < size; i++)
        {
            if(i == 0)
            {
                answer = nums[i];
            }
            
            else
            {
                answer = answer ^ nums[i];
            }
        }
        
        return answer;
        
    }
};

File: SingleNumber.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        
        a = {}
        
        for num in nums:
            
            if num in a:
                
                a[num] = a[num] + 1
            
            else:
                
                a[num] = 1
                
                
                
        for key, value in a.items():
            
            if value == 1:
                
                return key
        
        
            
            
        